api/configuration/v1beta1/kongupstreampolicy_types.go

```diff
@@ -99,19 +101,26 @@ type KongUpstreamPolicySpec struct {
 
 	// Healthchecks defines the health check configurations in Kong.
 	Healthchecks *KongUpstreamHealthcheck `json:"healthchecks,omitempty"`
+
+	// StickySessions defines the sticky session configuration for the upstream.
+	// When enabled, clients will be routed to the same backend target based on a cookie.
+	// This requires Kong Enterprise Gateway and sets hash_on to "none".
+	//
+	// +optional
+	StickySessions *KongUpstreamStickySessions `json:"stickySessions,omitempty"`
 }
 
 // HashInput is the input for consistent-hashing load balancing algorithm.
-// Can be one of: "ip", "consumer", "path".
-// +kubebuilder:validation:Enum=ip;consumer;path
+// Use "none" to disable hashing, it is required for sticky sessions.
+// +kubebuilder:validation:Enum=ip;consumer;path;none
 // +apireference:kic:include
 type HashInput string
 
 // KongUpstreamHash defines how to calculate hash for consistent-hashing load balancing algorithm.
 // Only one of the fields must be set.
 // +apireference:kic:include
 type KongUpstreamHash struct {
-	// Input allows using one of the predefined inputs (ip, consumer, path).
+	// Input allows using one of the predefined inputs (ip, consumer, path, none).
```

Comment:
```suggestion
	// Input allows using one of the predefined inputs (ip, consumer, path, none).
	// Set this to `none` or leave unset if you want to use sticky sessions.
```

Reply:
@pmalek Since "none" is hard-coded in the implementation of Gateway, I believe that not setting it is not equivalent to setting it to "none." 

Although we can add specific processing logic for it in the KIC implementation, keeping it consistent with the Gateway may provide a better user experience and reduce understanding costs. WDYT?

Reply:
> > @pmalek Since "none" is hard-coded in the implementation of Gateway, I believe that not setting it is not equivalent to setting it to "none."
> 
> Can we verify that somehow? Is `none` the default? Or it needs to be explicitly set?

If not set, it will be `nil` instead of `none` .  I think this was hard-coded by the gateway team during implementation to avoid introducing more complexity.

```
if (hash_on == "none" or hash_on == nil or hash_on == null) and
     (sticky_sessions_cookie == nil or sticky_sessions_cookie == null) then
    return -- not hashing, exit fast
  end
```

link: https://github.com/Kong/kong-ee/pull/12524/files#diff-6261fd809f9443ed32af2bb25d0f97266603688f70de2dec187f5d6f9ab36fe2R117

Reply:
> `hash_on == nil or hash_on == null`

Lua 🤯 

---

From the logic that you linked it would seem that early exit is done with `hash_on` is either of `none` or unset but I'm OK moving forward with relying only on the explicit value if that makes sense for you as well:

```suggestion
	// Input allows using one of the predefined inputs (ip, consumer, path, none).
	// Set this to `none` if you want to use sticky sessions.
```

WDYT?

Reply:
Great!